### PR TITLE
piwww: Naming refactor and cleanup

### PR DESCRIPTION
**Pull request by thi4go**
This diff modifies piwww user commands to start with the `user` prefix. It also removes request/response from help messages and unexports the command struct of the `piwww/` folder files.

### CODE REVIEW

File: politeiawww/cmd/piwww/userverifypayment.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import "github.com/decred/politeia/politeiawww/cmd/shared"
+
+// userVerifyPaymentCmd checks on the status of the logged in user's
+// registration payment.
+type userVerifyPaymentCmd struct{}
+
+// Execute executes the verify user payment command.
+func (cmd *userVerifyPaymentCmd) Execute(args []string) error {
+	vupr, err := client.VerifyUserPayment()
```
Comment (thi4go) [inline politeiawww/cmd/piwww/userverifypayment.go]: Client method namings are consistent with the types `www` api describes, so I thought it was better to just leave them to match the types.

File: politeiawww/cmd/piwww/piwww.go [hunk 2]
```diff
@@ -45,62 +45,67 @@ type piwww struct {
...
-	ProposalEdit      ProposalEditCmd      `command:"proposaledit"`
-	ProposalSetStatus ProposalSetStatusCmd `command:"proposalsetstatus"`
-	Proposals         ProposalsCmd         `command:"proposals"`
-	ProposalInventory ProposalInventoryCmd `command:"proposalinventory"`
+	ProposalNew       proposalNewCmd       `command:"proposalnew"`
+	ProposalEdit      proposalEditCmd      `command:"proposaledit"`
+	ProposalSetStatus proposalSetStatusCmd `command:"proposalsetstatus"`
+	Proposals         proposalsCmd         `command:"proposals"`
+	ProposalInventory proposalInventoryCmd `command:"proposalinventory"`
+	ProposalsVetted   proposalsVettedCmd   `command:"proposalsvetted" description:"(public) get a page of vetted proposals"`
```
Comment (lukebp) [inline politeiawww/cmd/piwww/piwww.go]: why is there a seperate command for vetted proposals?

File: politeiawww/cmd/piwww/piwww.go [hunk 3]
```diff
@@ -45,62 +45,67 @@ type piwww struct {
...
+	VoteInventory   voteInventoryCmd   `command:"voteinventory" description:"(public) retrieve the tokens of all public, non-abandoned proposal separated by vote status"`
+	// XXX www vote routes
+	VoteDetails  voteDetailsCmd  `command:"votedetails" description:"(public) get the details for a proposal vote"`
+	VoteStatus   voteStatusCmd   `command:"votestatus" description:"(public) get the vote status of a proposal"`
+	VoteStatuses voteStatusesCmd `command:"votestatuses" description:"(public) get the vote status for all public proposals"`
+
+	// User commands
+	UserDetails        userDetailsCmd          `command:"userdetails" description:"(public) get the details of a user profile"`
+	UserEdit           userEditCmd             `command:"useredit" description:"(user) edit the preferences of the logged in user"`
+	UserNew            userNewCmd              `command:"usernew" description:"(public) create a new user"`
```
Comment (lukebp) [inline politeiawww/cmd/piwww/piwww.go]: Can we put the ordering as:
```
UserNew
UserEdit
UserDetails
```

This follows the same convention as the other routes.  Routes that write data are listed first.  Routes that read data are listed afterwards.

File: politeiawww/cmd/piwww/piwww.go [hunk 4]
```diff
@@ -45,62 +45,67 @@ type piwww struct {
...
+	UserPaymentsRescan userPaymentsRescanCmd   `command:"userpaymentsrescan" description:"(admin) rescan a user's payments to check for missed payments"`
+	UserPendingPayment userPendingPaymentCmd   `command:"userpendingpayment" description:"(user) get details for a pending payment for the logged in user"`
+	UserProposals      userProposalsCmd        `command:"userproposals" description:"(public) get all proposals submitted by a specific user"`
+	UserVerifyEmail    userVerifyEmailCmd      `command:"userverifyemail" description:"(public) verify a user's email address"`
+	UserVerifyPayment  userVerifyPaymentCmd    `command:"userverifypayment" description:"(user) check if the logged in user has paid their user registration fee"`
+	Users              shared.UsersCmd         `command:"users" description:"(public) get a list of users"`
```
Comment (lukebp) [inline politeiawww/cmd/piwww/piwww.go]: The `Users` command should come after all the singular `User` commands.

File: politeiawww/cmd/piwww/piwww.go [hunk 5]
```diff
@@ -45,62 +45,67 @@ type piwww struct {
...
-	ManageUser         shared.ManageUserCmd     `command:"manageuser" description:"(admin)  edit certain properties of the specified user"`
-	Me                 shared.MeCmd             `command:"me" description:"(user)   get user details for the logged in user"`
-	NewUser            NewUserCmd               `command:"newuser" description:"(public) create a new user"`
-	Policy             PolicyCmd                `command:"policy" description:"(public) get the server policy"`
-	ProposalPaywall    ProposalPaywallCmd       `command:"proposalpaywall" description:"(user)   get proposal paywall details for the logged in user"`
-	RescanUserPayments RescanUserPaymentsCmd    `command:"rescanuserpayments" description:"(admin)  rescan a user's payments to check for missed payments"`
-	ResendVerification ResendVerificationCmd    `command:"resendverification" description:"(public) resend the user verification email"`
+	Me                 shared.MeCmd             `command:"me" description:"(user) get user details for the logged in user"`
+	ChangePassword     shared.ChangePasswordCmd `command:"changepassword" description:"(user) change the password for the logged in user"`
```
Comment (lukebp) [inline politeiawww/cmd/piwww/piwww.go]: Let's make these command names consistent with the www routes.  
https://github.com/lukebp/politeia/blob/tlog/politeiawww/api/www/v1/v1.go#L31-L51

I think the one exception should be the me command.  So `login`, `logout`, and `me` will all be kept the same.  The other user commands will start with User.

Examples:
```
UserPasswordChange
UserUsernameChange
UserTOTPSet
UserTOTPVerify
...
```

File: politeiawww/cmd/piwww/piwww.go [hunk 6]
```diff
@@ -45,62 +45,67 @@ type piwww struct {
...
+	VoteStatus   voteStatusCmd   `command:"votestatus" description:"(public) get the vote status of a proposal"`
+	VoteStatuses voteStatusesCmd `command:"votestatuses" description:"(public) get the vote status for all public proposals"`
+
+	// User commands
+	UserDetails        userDetailsCmd          `command:"userdetails" description:"(public) get the details of a user profile"`
+	UserEdit           userEditCmd             `command:"useredit" description:"(user) edit the preferences of the logged in user"`
+	UserNew            userNewCmd              `command:"usernew" description:"(public) create a new user"`
+	UserPaymentsRescan userPaymentsRescanCmd   `command:"userpaymentsrescan" description:"(admin) rescan a user's payments to check for missed payments"`
+	UserPendingPayment userPendingPaymentCmd   `command:"userpendingpayment" description:"(user) get details for a pending payment for the logged in user"`
+	UserProposals      userProposalsCmd        `command:"userproposals" description:"(public) get all proposals submitted by a specific user"`
```
Comment (lukebp) [inline politeiawww/cmd/piwww/piwww.go]: The www `UserProposals` command is being deprecated.  This command will be deleted.  The pi proposals route will eventually allow filtering by user ID, but not for the intial release.

**Review by lukebp — CHANGES_REQUESTED** [review]

File: politeiawww/cmd/piwww/piwww.go (hunk 2 above)
Comment (thi4go) [reply politeiawww/cmd/piwww/piwww.go]: It was already there so I thought of leaving it, but indeed it doesn't make sense anymore to have that cmd. Removing it

**Review by lukebp — APPROVED** [review]

**Pull request merged**